addons/account/static/src/components/char_with_placeholder_field/char_with_placeholder_field.js

```diff
@@ -0,0 +1,52 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { exprToBoolean } from "@web/core/utils/strings";
+import { registry } from "@web/core/registry";
+import { CharField, charField } from "@web/views/fields/char/char_field";
+import { formatChar } from "@web/views/fields/formatters";
+
+export class CharWithPlaceholderField extends CharField {
+    static template = "account.CharWithPlaceholderField";
+    static props = {
+        ...CharField.props,
+        placeholderField: { type: String },
+    };
+
+    /** Override **/
+    get formattedValue() {
+        const value =
+            this.props.record.data[this.props.name] ||
+            this.props.record.data[this.props.placeholderField];
+        return formatChar(value, { isPassword: this.props.isPassword });
+    }
+
```

Comment:
Not needed with the `placeholder` getter, right?
```suggestion
```

Reply:
This (as well as the code you commented on below) is because the list renderer creates lots of Fields with `props.readonly = true`.
The `readonly` is set to `false` only when the user clicks on the field to edit it.

https://github.com/odoo/odoo/blob/505a5ca54f8f4e204352b69d4388786c20933181/addons/web/static/src/views/fields/char/char_field.xml#L5-L7